### PR TITLE
Improve gRPC exception flow and trace readibility

### PR DESCRIPTION
**Pull request by ggezer**
- Rethrow `UnavailableException` coming form auth-layer to avoid confusion.
- Simplify channel traces by avoiding Optional field representation(`Optional["value"]`)

### CODE REVIEW

**Comment by ggezer** [comment]
@calvinjia PTAL. Thanks!

File: core/common/src/main/java/alluxio/grpc/GrpcChannelBuilder.java [hunk 1]
```diff
@@ -236,6 +237,11 @@ public GrpcChannel build() throws AlluxioStatusException {
       // Release the managed channel to the pool before throwing.
       GrpcManagedChannelPool.INSTANCE().releaseManagedChannel(mChannelKey,
           mConfiguration.getMs(PropertyKey.NETWORK_CONNECTION_SHUTDOWN_TIMEOUT));
+      if (exc instanceof UnavailableException) {
+        // Override exception from authentication layer.
+        throw new UnavailableException(
+            String.format("Target Unavailable. %s", mChannelKey.toStringShort()));
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/grpc/GrpcChannelBuilder.java]: Is there any info we want to retain from the original exception (ie stack trace?)

File: core/common/src/main/java/alluxio/grpc/GrpcChannelKey.java [hunk 2]
```diff
@@ -272,14 +272,14 @@ public boolean equals(Object other) {
   @Override
   public String toString() {
     return MoreObjects.toStringHelper(this)
-        .add("ChannelId", mChannelId)
-        .add("ClientType", mClientType)
+        .add("ClientType", getStringFromOptional(mClientType))
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/grpc/GrpcChannelKey.java]: Should we add `omitNullValues`?

Comment (ggezer) [reply core/common/src/main/java/alluxio/grpc/GrpcChannelKey.java]: That's very nice!

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Pull request merged**